docs/pages/developer-guide.md

```diff
@@ -936,6 +936,160 @@ In the example above, a server task can be accessed through the address:
 server-lb.hello-world.l4lb.thisdcos.directory:80
 ```
 
+# TLS
+
+**This feature works only on `Mesosphere DC/OS Enterprise` and is not supported on `DC/OS open`.**
+
+The SDK provides automated way of provisioning X.509 certificates and private keys referered in this documentation also as TLS artifacts. TLS artifacts can be used to expose service over encrypted TLS connection.
+
+One or more TLS artifacts can be requrested by adding `transport-encryption` key to the `task` YAML definition.
+
+```yaml
+name: "hello-world"
+pods:
+  hello:
+    count: 1
+    tasks:
+      server:
+        goal: RUNNING
+        cmd: "./service-with-tls --private-key=tls.key --certificate=tls.crt --ca-bundle=tls.ca"
+        cpus: 0.2
+        memory: 256
+        transport-encryption:
+          - name: tls
+            type: TLS
+        ports:
+            http:
+                protocol: tcp
+                port: 8080
+                vip:
+                    prefix: server-lb
+                    port: 80
+```
+
+Every item in the `transport-encryption` configuration section must have unique `name`. The `type` represents a format in which the private key and certificate will be delivered into the task sandbox. Different items will result in a unique private key and certificate.
+
+The TLS artifacts within single task will be shared across all instances of the pod task. Different tasks can request TLS artficats with same name but each task will get unique private key and certificate and they won't get shared across different tasks.
+
+In a given example the `$MESOS_SANDBOX` directory would contain following files:
+
+```
+$MESOS_SANDBOX/
+               ...
+               tls.key
+               tls.crt
+               tls.ca
+               ...
+```
```

Comment:
Looking at it this way, the type "tls" should probably have been called "PEM" or "OpenSSL" :-).